src/main/java/spoon/Launcher.java

```diff
@@ -125,12 +125,28 @@ public void printUsage() {
 	}
 
 	/**
-	 * Constructor with no arguments.
+	 * Creates a {@link Launcher} using the {@link Factory} returned by {@link #createFactory()}.
 	 */
 	public Launcher() {
+		factory = createFactory();
 		processArguments();
 	}
 
+	/**
+	 * Creates a {@link Launcher} with {@link Factory} {@code pFactory}.
+	 *
+	 * @param pFactory
+	 * 		The {@link Factory} that will be utilized in {@link #buildModel()}.
+	 * @throws IllegalArgumentException
+	 * 		If {@code pFactory == null}.
+	 */
+	public Launcher(final Factory pFactory) {
+		if (pFactory == null) {
+			throw new IllegalArgumentException("unable to create launcher with null factory");
+		}
+		factory = pFactory;
```

Comment:
You can also call this constructor from the previous one ```this(createFactory())``` in order to reduce code duplication.